src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,22 +1,30 @@
-import {Component} from '@angular/core';
-import {productsMock} from '../../../shared/products/products.mock';
+import {Component, Input, OnInit, Output, EventEmitter} from '@angular/core';
+import {IProduct} from '../../../shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {
-    readonly product = productsMock[0];
+export class CardComponent implements OnInit {
+    @Input() card: IProduct | null = null;
+    @Output() addToCart = new EventEmitter<string>();
+    rating = 0;
 
-    onProductBuy(event: Event) {
-        event.stopPropagation();
+    // Todo: Я помню ты говорил что надо через родителя, а такой вариант не подходит?
+    ngOnInit(): void {
+        if (this.card) {
+            this.rating = this.card.rating;
+        }
+    }
 
-        // eslint-disable-next-line no-console
-        console.log('Buy product');
+    onAddToCart(): void {
+        if (this.card) {
+            this.addToCart.emit(this.card?._id);
```

Comment:
У тебя выше идет проверка на наличие `this.card`, по этому `?.` внутри `if` не нужен
```suggestion
            this.addToCart.emit(this.card._id);
```

Reply:
Поправил